### ISSUE TITLE
Bug Report | Connection timeout in Firefox for non-root paths

### ISSUE THREAD

**Issue by spikecodes** [http-rs/tide#785]
I'm building a project using Tide and in my project, the response is returned after several seconds. In Firefox, this causes a "connection timeout". Though, I feel this is a tide error because the issue does *not* occur when the route's path is `/`, only when it it a non-root path.

To reproduce:

1. Run:
```rust
use tide::Request;

#[async_std::main]
async fn main() -> tide::Result<()> {
	let mut app = tide::new();
	app.at("/test").get(test);
	app.listen("127.0.0.1:8070").await?;
	Ok(())
}

async fn test(_req: Request<()>) -> tide::Result {
	async_std::task::sleep(std::time::Duration::from_secs(6)).await;
	Ok("a".into())
}
```
2. Visit http://localhost:8070/test in Firefox
3. See error:
![image](https://user-images.githubusercontent.com/19519553/105793077-63d09a00-5f80-11eb-8f79-f19d08b9ba27.png)

This issue does *not* occur when the path is the root (when it has no segments other than `/`):
```rust
app.at("/").get(test);
```
![image](https://user-images.githubusercontent.com/19519553/105882691-dc297080-5ffd-11eb-9e28-b6274e86a8a2.png)

**Comment by jbr** [comment]
I wonder if this has to do with the fact that async-h1 doesn't properly parse the keep-alive header

**Comment by pepoviola** [comment]
I will try to reproduce from the snippet, and see if I found some hints.
Thx!

**Comment by spikecodes** [comment]
Thanks! Let me know if you find anything or are able to reproduce

**Comment by spikecodes** [comment]
Fixed by Tide v0.16.0!
